go/vt/concurrency/error_group.go

```diff
@@ -19,60 +19,80 @@ package concurrency
 import "context"
 
 // ErrorGroup provides a function for waiting for N goroutines to complete with
-// at least X successes and no more than Y failures, and cancelling the rest.
+// at least Z successes which we wanted to wait for and
+// at least X overall successes and no more than Y failures, and cancelling the rest.
 //
 // It should be used as follows:
 //
-//		errCh := make(chan error)
+//		errCh := make(chan concurrency.Error)
 //		errgroupCtx, errgroupCancel := context.WithCancel(ctx)
 //
 //		for _, arg := range args {
 //			arg := arg
 //
 //			go func() {
 //				err := doWork(errGroupCtx, arg)
-//				errCh <- err
+//				errCh <- concurrency.Error{
+//					Err: err,
+//					MustWaitFor: <boolean>,
+//				}
 //			}()
 //		}
 //
 //		errgroup := concurrency.ErrorGroup{
 //			NumGoroutines: len(args),
 //			NumRequiredSuccess: 5, // need at least 5 to respond with nil error before cancelling the rest
 //			NumAllowedErrors: 1, // if more than 1 responds with non-nil error, cancel the rest
+// 			NumErrorsToWaitFor: 1, // if there is 1 response that we must wait for, before cancelling the rest
 //		}
 //		errRec := errgroup.Wait(errgroupCancel, errCh)
 //
 //		if errRec.HasErrors() {
 //			// ...
 //		}
+//
+//	The NumErrorsToWaitFor should be equal to the number of
+//	Errors that are received on the channel which have MustWaitFor set
 type ErrorGroup struct {
 	NumGoroutines        int
 	NumRequiredSuccesses int
 	NumAllowedErrors     int
+	NumErrorsToWaitFor   int
+}
+
+// Error is used in ErrGroup.Wait function
+// It contains the error that was received along
+// with the information of whether the received error
+// originated from a tablet that we must wait for
+type Error struct {
```

Comment:
I don't know whether it is good practice to use the same name as a builtin type, but just exported instead of not..
@sougou @vmg WDYT?

Reply:
I thought of this for a while, but couldn't come up with a meaningful name for it. Looking for suggestions. The ones that I considered where `SpecialError`, `ErrorGroupError`, `MustWaitForError`. None of these alternatives seemed better than just `Error`

Reply:
That's perfectly OK, the builtin type is lowercase `error`.